dask/dataframe/shuffle.py

```diff
@@ -981,12 +982,28 @@ def compute_and_set_divisions(df, **kwargs):
         or any(a > b for a, b in zip(mins, maxes))
     ):
         raise ValueError(
-            "Partitions must be sorted ascending with the index", mins, maxes
+            f"Partitions must be sorted ascending by {column.name}", mins, maxes
+        )
+    return mins, maxes
+
+
+def compute_divisions(df, col=None, **kwargs) -> tuple:
+    mins, maxes = compute_mins_and_maxes(df, col=col, **kwargs)
+    has_overlap = any(mins[i] <= maxes[i - 1] for i in range(1, len(mins)))
+    if has_overlap:
+        raise ValueError(
+            f"Partitions must be sorted ascending by {'index' if col is None else col}"
         )
 
+    return tuple(mins) + (list(maxes)[-1],)
+
+
+def compute_and_set_divisions(df, **kwargs):
+    mins, maxes = compute_mins_and_maxes(df, **kwargs)
+
     df.divisions = tuple(mins) + (list(maxes)[-1],)
+    overlap = [i for i in range(1, len(mins)) if mins[i] <= maxes[i - 1]]
```

Comment:
Why'd you flip the `>=` to `<=` here? I think what you have now is right, but I'm not sure if `fix_overlap` is expecting the inverse for some reason.

Reply:
Thanks for reminding me! Yeah I think it was wrong before and sending too much to `fix_overlap`. But I need to double check that theory.